packages/frontend/src/components/FlowStepTestController/index.tsx

```diff
@@ -95,6 +128,7 @@ export default function FlowStepTestController(
     return validateStepParams(step, testExecutionSteps, substeps)
   }, [testExecutionSteps, step, substeps])
 
+  const shouldAllowCheckStep = !(!isValid || readOnly || isSaving)
```

Comment:
sorz is it me but i find it quite hard to understand this: this is equivalent to `isValid && !readOnly && !isSaving` right?